airflow/cli/commands/task_command.py

```diff
@@ -42,9 +45,27 @@
     get_dags,
     suppress_logs_and_warning,
 )
+from airflow.utils.dates import timezone
 from airflow.utils.log.logging_mixin import StreamLogWriter
 from airflow.utils.net import get_hostname
-from airflow.utils.session import create_session
+from airflow.utils.session import create_session, provide_session
+
+
+def _get_ti(dag, task, exec_date_or_run_id):
```

Comment:
Do we need to pass `dag` here -- isn't that available as `task.dag`?

Reply:
Or alrternatively, do we need to pass `task` here? Passing both seems unnecessary